### ISSUE TITLE
Items despawn in mine carts

### ISSUE THREAD

**Issue by AlexYst** [minetest/minetest_game#2717]
Items held in mine carts despawn like items dropped in the world. Is this intentional?

Consider that you've gone down to your mine, which is linked to the surface by a powered rail line. You've brought with you a stack of mine carts, so every time your inventory fills up, you drop everything in a cart, send it on it's way, then keep mining. You get down to your last cart, so you fill up your inventory one last time and ride that cart to the surface, only to find all your previous carts are waiting at home empty.

It seems like this would be one of the primary use cases for putting items in a cart, but it doesn't really work for that. You also can't move items vast distances between builds using mine carts for the same reason.

**Comment by sfan5** [comment]
This is not a bug in the sense that it has always been like this, but it's obvious that this behavior is not good.

**Comment by An0n3m0us** [comment]
The items are not lost; they seem to fall out of the cart around 67-71 blocks away from the player (measured on flat landscape; probably varies with height).
So it could be related to chunk loading?

Also, the cart stops moving at that distance, until the player approaches it.

**Comment by AlexYst** [comment]
No, the items are definitely lost. In one of my experiments, I sent a cart of items a great distance and they might have been dropped before being despawned (I've been having problems with the \*player\* getting dropped by the cart too), but I rode in the cart with the items. They were never any number of blocks away. In another experiment, I put a single piece of cobble in a cart, pushed it a few nodes to make the cart pick up the item, then just waited. The cobble despawned while still in the cart.

**Comment by orbea** [comment]
It would make sense to just support chests on carts to entirely avoid these problems.

**Comment by AlexYst** [comment]
I've actually been thinking something similar, though I was considering just adding an inventory to the regular carts. Adding the option to add a chest to a cart might be a better idea though.

**Comment by wsor4035** [comment]
if a chest cart is to be made, maybe it would be beneficial to provide a way to register different types of carts like you can register different types of rails already

**Comment by 56independent** [comment]
I've been having the same problem. When i build cart lines to expand the base, i end up with a lot of waste. So i send it back on a cart, but as i follow, the items suddenly drop onto the track and i end up having to pick them up manually. I've never had an instance where an item despawns; they just fall off.

However, my server does have these, if such information is relevant:

* Rubberbanding, heavy at times
* Long cart lines which are often unloaded
* A much higher cart speed
* Various cart mods:
    * `boost_cart`
    * `minecart`
